util.c

```diff
@@ -31,7 +31,8 @@ size_t max_filename_length(const char* path)
         PERROR("statfs in max_filename_length failed");
         return 255; // Seems to be a safe default.
     }
-    return (size_t)sfs.f_namelen;
+    size_t len = (size_t)sfs.f_namelen;
+    return len < 255 ? 255 : len;
```

Comment:
It was a s3ql fuse fs, on a debian stable machine.  For what I see, I think the bug was in this particular version of FUSE, not in s3ql.  Still, for what I read, there is many way to mess with this, so it's probably a good thing to do anyway.